src/sub.c

```diff
@@ -1124,6 +1124,54 @@ natsSubscription_QueuedMsgs(natsSubscription *sub, uint64_t *queuedMsgs)
     return s;
 }
 
+int64_t
+natsSubscription_GetID(natsSubscription* sub)
+{
+    if (sub == NULL)
+        return 0;
+
+    natsSub_Lock(sub);
+
+    if (sub->closed)
+    {
+        natsSub_Unlock(sub);
+        return 0;
+    }
+
+    int64_t id = 0;
+    SUB_DLV_WORKER_LOCK(sub);
```

Comment:
I missed that in earlier PRs, but for both these APIs, you don't need to call SUB_DLV_WORKER_LOCK, so let's remove.